src/sentry/api/serializers/models/event.py

```diff
@@ -176,8 +175,6 @@ def _get_user_report(self, user, event):
         return serialize(user_report, user)
 
     def get_attrs(self, item_list, user, is_public=False):
-        eventstore.bind_nodes(item_list, "data")
```

Comment:
Could you elaborate on how you can ensure that the event has been bound by the time we get here ? This is called at every serialization. Can we add a guard in the bind_nodes method to ensure the node is not bound twice ?
Making the assumption inside the serialization method seems quite dangerous.

Reply:
The weird thing is it's not actually necessary to ever manually bind nodes in order to get a correct result since NodeData will lazily fetch from nodestore on the first attempt to access `event.data`. (Weird as in, I don't know why we ever do it if not necessary and perhaps we should just remove them all now to prove it's not dangerous). The only place where this may have a performance difference is when we need to take advantage of nodestore's multi_get (the group_events, project_events and organization_events endpoints). I've updated these 3 methods to call `bind_nodes` directly.

Reply:
I think i've figured out the reason we have it is that it's required by the old Django Event model. SnubaEvent and eventstore.models.Event autoload data

Reply:
When saying that SnubaEvent and eventstore.Event autoload data, are you talking about this line ?
https://github.com/getsentry/sentry/blob/master/src/sentry/db/models/fields/node.py#L112

So if this is the case and the Django model does not do that and the Django model is still around, shouldn't this line still be needed ?

Reply:
Yes, that's the one. So there are 2 remaining places in the codebase where we might have a Django Event. One is in the event manager https://github.com/getsentry/sentry/blob/master/src/sentry/event_manager.py#L394-L401 and the other is https://github.com/getsentry/sentry/blob/master/src/sentry/eventstream/kafka/protocol.py#L37-L50. In both cases we have manually bound data so I don't think that we ever want to rebind in those cases. I suspect that we were rebinding unnecessarily in some of the post process tasks previously.